### PR TITLE
Stanford student work facet.

### LINKED ISSUE #151: New Stanford dissertations and theses facets
With facets for different degrees following the degree mapping on this ticket:
https://jirasul.stanford.edu/jira/browse/SW-1708

### PR DESCRIPTION
**Pull request by shelleydoljack**
Fixes #151 #157 #158 

@ndushay Please review and approve. I am awaiting feedback from testers by Feb 9 and plan to merge it by then.

### CODE REVIEW

File: stanford-sw/sw_index.properties [hunk 1]
```diff
@@ -84,7 +84,7 @@ vern_subject_other_subvy_search = custom, getLinkedField(600vy:610vy:611vy:630vy
 subject_all_search = custom, getAllAlphaSubfields(600:610:611:630:648:650:651:652:653:654:655:656:657:658:662:690:691:696:697:698:699)
 vern_subject_all_search = custom, getLinkedField(600[a-z]:610[a-z]:611[a-z]:630[a-z]:648[a-z]:650[a-z]:651[a-z]:652[a-z]:653[a-z]:654[a-z]:655[a-z]:656[a-z]:657[a-z]:658[a-z]:662[a-z]:690[a-z]:691[a-z]:696[a-z]:697[a-z]:698[a-z]:699[a-z])
 # Subject Facet Fields
-topic_facet = custom, getTopicWithoutTrailingPunct(600abcdq:600t:610ab:610t:630a:630t:650a, [\\\\,;:], ([\\p{L}\\p{N}]{4}|[\\)]) )
+topic_facet = custom, getTopicWithoutTrailingPunct(600abcdq:600t:610ab:610t:630a:630t:650a, [\\\\,;:], ([\\p{L}\\p{N}]{4}|[A-Za-z]{3}|[\\)]) )
```
Comment (ndushay) [inline stanford-sw/sw_index.properties]: looks fine, but no concomitant test?

File: stanford-sw/src/edu/stanford/StanfordIndexer.java [hunk 2]
```diff
@@ -2020,16 +2020,17 @@ private void setFundFacet(final Record record) {
  }
 
 /**
-* set locations based upon the value in the 852 subfield c or 999 subfield t
-* "Art Locked Stacks"
+* set locations based upon the value in the 852 subfield c or 999 subfield l
```
Comment (ndushay) [inline stanford-sw/src/edu/stanford/StanfordIndexer.java]: was the original comment wrong (subfield t) or is the new comment wrong (subfield l) or neither?

File: stanford-sw/test/src/edu/stanford/LocationTestsCurriculum.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.marc4j.marc.*;
+import org.xml.sax.SAXException;
+
+/**
+ * junit4 tests for location facet for locked stacks
+ * @author Shelley Doljack
+ */
+public class LocationTestsCurriculum extends AbstractStanfordTest
```
Comment (ndushay) [inline stanford-sw/test/src/edu/stanford/LocationTestsCurriculum.java]: it's actually a JUnit practice to call test classes xxxTests.   https://blog.testproject.io/2017/04/16/test-automation-best-practices/

File: stanford-sw/test/src/edu/stanford/LocationTestsLockedStacks.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
  * junit4 tests for location facet for locked stacks
  * @author Laney McGlohon
  */
-public class LockedStacksLocationTests extends AbstractStanfordTest
+public class LocationTestsLockedStacks extends AbstractStanfordTest
```
Comment (ndushay) [inline stanford-sw/test/src/edu/stanford/LocationTestsLockedStacks.java]: ditto on class name

File: .travis.yml [hunk 5]
```diff
@@ -4,8 +4,17 @@ env:
...
+    packages:
+      - openjdk-6-jdk
+      - ant
+      - ant-optional
 jdk:
-  - openjdk7
+  # prod, test: Java HostSpot 64-bit Server VM;
+  #     Java(TM) SE Runtime Environment (build 1.6.0_33-b03)
+  - openjdk6
+#  - openjdk7  # this was being tested, but now causes failures
```
Comment (ndushay) [inline .travis.yml]: you figured this out before but it never made it to a PR?  huh.  Glad to see it, in any case.

File: stanford-sw/sw_index.properties [hunk 6]
```diff
@@ -1,195 +1,198 @@
...
+callnum_facet_hsim = custom, getCallNumHierarchVals(|, callnumber_map)
+callnum_search = custom, getLocalCallNums
+shelfkey = custom, getShelfkeys
+reverse_shelfkey = custom, getReverseShelfkeys
+
+# Location facet
+location_facet = custom, getLocationFacet
+
+# Stanford theses and dissertations facet
+stanford_theses_facet_hsim = custom, getStanfordThesesFacet
```
Comment (ndushay) [inline stanford-sw/sw_index.properties]: totally not worth redoing, but this change should have gone with the commit for the relevant code changes.  This commit is now really doing 2 changes, but the description only mentions one.

**Review by ndushay — CHANGES_REQUESTED** [review]
Shelley,

Looks good.  A number of minor possible changes:

- no test for topic facet change (maybe that's ok?)
- a possible comment typo for `setLocationFacet`
- names of new test classes don't follow "best practice"

In a perfect world, for review practices, it's easier if the whitespace changes are a separate commit from code changes -- I know this code base is a mess for whitespace, tho.

File: stanford-sw/src/edu/stanford/StanfordIndexer.java (hunk 2 above)
Comment (shelleydoljack) [reply stanford-sw/src/edu/stanford/StanfordIndexer.java]: The original comment was wrong.

**Review by ndushay — APPROVED** [review]

**Pull request merged**